server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailDispatcher.java

```diff
@@ -130,8 +140,8 @@ public void dispatch(Mail mail) throws MessagingException {
                 Map<String, List<String>> savedHeaders = saveHeaders(mail, recipient);
 
                 addSpecificHeadersForRecipient(mail, message, recipient);
-                mailStore.storeMail(recipient, mail);
-                
+                storeMailWithRetry(mail, recipient).block();
```

Comment:
If we are about to use reactor, can t we achieve all delivery in a flux with concatMap?

Reply:
I had a look but as the dispatcher mutate quite a bunch of states, it looks dangerous to use a reactive pipeline here